### PR TITLE
Restore OpenAPI specification for HTTP API (DB-197)

### PR DESCRIPTION
**Pull request by pvanbuijtene**
Added: Restored OpenAPI specification for HTTP API.

Restored from [release/oss-v5/docs/http-api/swagger.yaml](https://github.com/EventStore/EventStore/blob/release/oss-v5/docs/http-api/swagger.yaml)

### CODE REVIEW

File: docs/http-api/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,2495 @@
...
+        Scavenge reclaims disk space by rewriting database chunks, minus the
+        events to delete, and then deleting the old chunks.
+      summary: Scavenge a node
+      tags:
+        - Admin
+      operationId: Scavenge a node
+      produces:
+        - application/json
+      parameters:
+        - name: startFromChunk
```
Comment (timothycoleman) [inline docs/http-api/swagger.yaml:95]: this is ignored now

File: docs/http-api/swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,2495 @@
...
+          default: 1
+        - name: threshold
+          in: query
+          description: The threshold to use during the scavenge.
+          type: integer
+          required: false
+          default: 0
+        - name: throttlePercent
+          in: query
+          description: The amount of throttle to use during the scavenge.
```
Comment (timothycoleman) [inline docs/http-api/swagger.yaml:114]: > When set to 100 the scavenge will run at full speed. When set to 50 the scavenge will take twice as long by pausing at regular intervals. This allows it to be run with less impact on the server.

File: docs/http-api/swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,2495 @@
...
+          type: integer
+          required: false
+          default: 0
```
Comment (timothycoleman) [inline docs/http-api/swagger.yaml:117]: 100

File: docs/http-api/swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,2495 @@
...
+  /metrics:
+    get:
+      description: Returns the metrics of a node in OpenMetrics format.
+      summary: Node metrics
+      tags:
+        - Stats
+      operationId: Node metrics
+      parameters: []
+      produces:
+        - application/openmetrics-text
```
Comment (timothycoleman) [inline docs/http-api/swagger.yaml:233]: should this be the actual content-type header of the response? if so it looks like it is text/plain

**Review by timothycoleman — DISMISSED** [review]
thanks, suggestions/questions in line :+1:

File: docs/http-api/swagger.yaml (hunk 4 above)
Comment (pvanbuijtene) [reply docs/http-api/swagger.yaml]: hmm we specify it in the controller like that, but see it's not used indeed.

File: docs/http-api/swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,2496 @@
...
+          required: true
+          type: string
+        - name: subscription
+          in: path
+          description: The name of the subscription group
+          required: true
+          type: string
+        - name: embed
+          in: query
+          description: Needed
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:692]:
```suggestion

```

File: docs/http-api/swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,2496 @@
...
+          required: true
+          type: string
+        - in: body
+          name: subscriptionItem
+          description: Subscription to create
+          required: false
+          schema:
+            $ref: "#/definitions/SubscriptionItem"
+      responses:
+        "200":
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:768]:
```suggestion
        "201":
```
Creating a persistent subscription can also return:
- 209: conflict
- 401: unauthorized
- 400: bad request

File: docs/http-api/swagger.yaml [hunk 7]
```diff
@@ -0,0 +1,2496 @@
...
+          required: true
+          type: string
+        - name: messageid
+          in: path
+          description: The id of the message that needs to be acked
+          required: true
+          type: string
+      responses:
+        "200":
+          description: New persistant subscription
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:871]:
```suggestion
          description: message was acked
```

File: docs/http-api/swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,2496 @@
...
+          required: true
+          type: string
+        - name: ids
+          in: query
+          description: The ids of the messages that need to be acked separated by commas
+          required: false
+          type: string
+      responses:
+        "200":
+          description: New persistant subscription
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:907]:
```suggestion
          description: messages were acked
```

File: docs/http-api/swagger.yaml [hunk 9]
```diff
@@ -0,0 +1,2496 @@
...
+          description: >-
+            <ul><li>**Park** - Don't retry the message, park it until a request
+            is sent to reply the parked messages<li>**Retry** - Retry the
+            message<li>**Skip** - Discard the message<li>**Stop** - Stop the
+            subscription</ul>
+          required: false
+          type: string
+          enum:
+            - Park
+            - Retyr
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:952]:
```suggestion
            - Retry
```

File: docs/http-api/swagger.yaml [hunk 10]
```diff
@@ -0,0 +1,2496 @@
...
+            - Skip
+            - Stop
+      responses:
+        "200":
+          description: New persistant subscription
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:957]:
```suggestion
          description: message was nacked
```

File: docs/http-api/swagger.yaml [hunk 11]
```diff
@@ -0,0 +1,2496 @@
...
+          required: false
+          type: string
+          enum:
+            - Park
+            - Retry
+            - Skip
+            - Stop
+      responses:
+        "200":
+          description: New persistant subscription
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:1007]:
```suggestion
          description: messages were nacked
```

File: docs/http-api/swagger.yaml [hunk 12]
```diff
@@ -0,0 +1,2496 @@
...
+      summary: Get all users
+      description: Returns all users defined in EventStoreDB.
+      operationId: Get all users
+      produces:
+        - application/json
+        - application/xml
+      responses:
+        "200":
+          description: OK
+        "404":
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:1069]:
```suggestion

```
This operation can't return not found

All operations in the UsersController can return 401 Unauthorized

File: docs/http-api/swagger.yaml [hunk 13]
```diff
@@ -0,0 +1,2496 @@
...
+          description: Not found
+    put:
+      tags:
+        - Users
+      summary: Update specified user
+      description: Update the FullName of Groups of the specified user.
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:1123]:
```suggestion
      description: Update the FullName or Groups of the specified user.
```

File: docs/http-api/swagger.yaml [hunk 14]
```diff
@@ -0,0 +1,2496 @@
...
+        - application/xml
+        - application/json
+      parameters:
+        - name: login
+          in: path
+          description: The user's name
+          required: true
+          type: string
+      responses:
+        "204":
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:1161]:
```suggestion
        "200":
```

**Review by hayley-jean — CHANGES_REQUESTED** [review]

**Review by ylorph — DISMISSED** [review]

File: docs/http-api/swagger.yaml [hunk 15]
```diff
@@ -0,0 +1,2792 @@
...
+      operationId: Return Gossip details
+      produces:
+        - application/json
+        - application/xml
+      responses:
+        "200":
+          description: OK
+        "404":
+          description: Not found
+    post:
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:2177]: There is no longer a `post` operation for gossip in the HTTP API

File: docs/http-api/swagger.yaml [hunk 16]
```diff
@@ -0,0 +1,2792 @@
...
+        - application/json
+      parameters:
+        - name: name
+          in: path
+          description: The projection to delete
+          required: true
+          type: string
+        - name: deleteStateStream
+          in: query
+          description: TBD
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:2102]:
```suggestion
          description: Whether the state stream should also be deleted
```

File: docs/http-api/swagger.yaml [hunk 17]
```diff
@@ -0,0 +1,2792 @@
...
+          required: false
+          type: boolean
+        - name: deleteCheckpointStream
+          in: query
+          description: TBD
+          required: false
+          type: boolean
+        - name: deleteEmittedStreams
+          in: query
+          description: TBD
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:2112]:
```suggestion
          description: Whether the emitted streams should also be deleted. Note that this only has an effect if 'track emitted streams' was enabled on the projection
```

File: docs/http-api/swagger.yaml [hunk 18]
```diff
@@ -0,0 +1,2792 @@
...
+          enum:
+            - JS
+        - name: enabled
+          in: query
+          description: Is the projection enabled
+          required: false
+          type: boolean
+      responses:
+        "201":
+          description: New user created
```
Comment (hayley-jean) [inline docs/http-api/swagger.yaml:1731]:
```suggestion
          description: New projection created
```

**Review by hayley-jean — CHANGES_REQUESTED** [review]

**Review by hayley-jean — APPROVED** [review]

**Review by ylorph — APPROVED** [review]

**Pull request merged**